src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/OffenderService.kt

```diff
@@ -640,11 +641,32 @@ class OffenderService(
     return AuthorisableActionResult.Success(documents)
   }
 
+  fun getDocumentsFromApDeliusApi(crn: String): AuthorisableActionResult<List<Document>> {
+    val documentsResult = apDeliusContextApiClient.getDocuments(crn)
+
+    val documents = when (documentsResult) {
+      is ClientResult.Success -> documentsResult.body
+      is ClientResult.Failure.StatusCode -> when (documentsResult.status) {
+        HttpStatus.NOT_FOUND -> return AuthorisableActionResult.NotFound()
+        HttpStatus.FORBIDDEN -> return AuthorisableActionResult.Unauthorised()
+        else -> documentsResult.throwException()
+      }
+
+      is ClientResult.Failure -> documentsResult.throwException()
+    }
+    return AuthorisableActionResult.Success(documents)
+  }
+
   fun getDocument(
     crn: String,
     documentId: String,
     outputStream: OutputStream,
-  ) = communityApiClient.getDocument(crn, documentId, outputStream)
+    isRetrieveDocumentFromApDelius: Boolean,
```

Comment:
I don't think this matters too much as it's a short-lived change, but for consistency with the getDocuments functions maybe there should be two functions?

* `getDocumentFromCommunityApi`
* `getDocumentFromDeliusApi`

Reply:
sound good, and this will only take 5 so will make this change. Thanks!